sway-core/src/control_flow_analysis/dead_code_analysis.rs

```diff
@@ -456,13 +456,14 @@ fn connect_struct_declaration(
         .iter()
         .map(|field| (field.name.clone(), graph.add_node(field.into())))
         .collect::<Vec<_>>();
-    // If this is a library or smart contract, and if this is public, then we want to connect the
+    // If this is a smart contract, or a library and the struct is public, then we want to connect the
```

Comment:
Wouldn't this mean that a struct that is only partially used in a contract will no longer generate a warning? The issue in #1305 is that a struct that is _returned_ from a contract should not generate a warning even if the contract does not directly accesses its fields.